webui/src/app/resolver/resolve-badges.ts

```diff
@@ -45,12 +47,25 @@ export class BadgesResolver implements Resolve<BadgeBean[]> {
         };
         this.badgesService.GetAllFromContent('', meta).subscribe(
             (data: ContentListResponse<BadgeBean>) => {
-                this.badgesStoreService.dispatch(
-                    new LoadBadgesAction({
-                        badges: data.content,
-                        metadata: data.pageMetadata,
-                    }, subject)
-                );
+                //var stats = {}
+                var i = 0
+                data.content.forEach(badge => {
+                    this.badgeStatsService.GetBadgeStats(badge.slug).subscribe(
+                        (stats: Map<string, number>) => {
+                            console.log("stats", stats);
+                            badge._stats = stats
+                            if (i == 3) {
```

Comment:
This is an awful hack because I do not know how to wait asynchronously until multiple calls (in a loop) finish. Maybe @sguiheux can help ?

Reply:
Maybe observableZip 
https://github.com/ovh/cds/blob/c762ace824e4cc451d066ee7fe6aa688d9631a90/ui/src/app/views/application/show/workflow/application.workflow.component.ts#L77

Reply:
It works! Thx @sguiheux